### PR TITLE
Add support for updating a previous message.

### PR DESCRIPTION
**Pull request by underscorebrody**
Resolves https://github.com/jenkinsci/slack-plugin/issues/583

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,6 +123,20 @@ slackSend(
...
 
+#### Update Messages
+
+You can update the content of a previously sent message using the pipeline step.
+The step returns an object which you can use to retrieve the timestamp
+
+Example:
+
+```groovy
+def slackResponse = slackSend(channel: "updating-stuff", message: "Here is the primary message")
```
Comment (Jetersen) [inline README.md]: why not support
```
slackResponse.updateMessage("yo dawg, I heard you like to update this message!")
```

File: src/main/java/jenkins/plugins/slack/StandardSlackService.java [hunk 2]
```diff
@@ -190,6 +193,13 @@ public boolean publish(SlackRequest slackRequest) {
                 json.put("unfurl_links", "true");
                 json.put("unfurl_media", "true");
 
+                String apiEndpoint = "chat.postMessage";
+
+                if (StringUtils.isNotEmpty(updateMessage)) {
+                    json.put("ts", updateMessage);
```
Comment (Jetersen) [inline src/main/java/jenkins/plugins/slack/StandardSlackService.java]: Call what it is, timestamp ⌚
```suggestion
                    json.put("ts", timestamp);
```

Comment (underscorebrody) [reply src/main/java/jenkins/plugins/slack/StandardSlackService.java]: Yup good point, i'll change it

File: README.md (hunk 1 above)
Comment (underscorebrody) [reply README.md]: I think that sounds like a good idea, though i'm not familiar enough with the ins-and-outs of jenkins plugins to be certain of the best way to implement it in a DRY fashion. Would you be able to give me any direction on how to best utilize the existing `SlackSendStep` within the `SlackResponse` class in a jenkins-friendly fashion?

Comment (steven-foster) [reply README.md]: I believe these response objects returned from pipeline steps are intended to be simple data objects without additional behaviour. Not 100% sure about that though!

Comment (Jetersen) [reply README.md]: Nothing prevents you from having simple data object still with methods on them :)
You can override the toString to print of the object nicely.

Comment (underscorebrody) [reply README.md]: Would it be possible to land this and then consider adding that method? As-is, this implementation has functionality that allows full usage of the feature following current patterns, and that additional method would be some short-cutting that would require a bit of dev effort.

**Comment by Jetersen** [comment]
One minor nit, I still think it would be helpful to add an help file: https://github.com/jenkinsci/slack-plugin/tree/master/src/main/resources/jenkins/plugins/slack/workflow/SlackSendStep

**Comment by underscorebrody** [comment]
@Jetersen done!

**Comment by timja** [comment]
@casz did you test it locally?

**Comment by underscorebrody** [comment]
FWIW I built and installed this on our jenkins instance and we are using it without issue.

**Comment by Jetersen** [comment]
@underscorebrody Thanks for your contribution Eric, much appreciated!

File: README.md (hunk 1 above)
Comment (Jetersen) [reply README.md]: Actually in #680 this was done with addReaction :)

**Pull request merged**